packages/flutter_tools/lib/src/fuchsia/fuchsia_device.dart

```diff
@@ -593,7 +630,8 @@ class FuchsiaDevice extends Device {
   bool _ipv6;
 
   /// [true] if the current host address is IPv6.
-  bool get ipv6 => _ipv6 ??= isIPv6Address(id);
+  //  Strip interface name to avoid issues
```

Comment:
Can this comment be more specific? Also @chaselatta, in light of https://github.com/flutter/flutter/pull/85341, I see that there are some sensitivities around the addresses that are used in different contexts, so I wanted to call this line to your attention for review.

Reply:
The problem is that dart's ipv6 parsing logic assumes that ipv6 address are hex numbers seperated by `:`. It is unable to accept an ipv6 address that is suffixed by an interface name, say `[::1%qemu]`. Therefore in order to avoid worksraound this inconsistency, I just strip that interface component of an ipv6 address away.

Reply:
Documented in 77e22ad50b0508d0b1682f7b9e437bb3fd224ae3, I will leave this unresolved in case you want more explaination or perharps another issue on dart sdk for this.